server/services/schedule-service.js

```diff
@@ -10,23 +11,14 @@ module.exports = {
     }
 
     const paymentDates = getPaymentDates()

```

Comment:
The `getPaymentDates` function returns 6 identical dates as the same date reference is updated each time, rather than creating a new one. You can create a new date object in the loop or clone an existing one via `new Date(date)`

Generally speaking for loops are a code smell in ES6, they can usually be replaced with map or reduce.

A more functional approach would be something like:
```
function getDate(dateIn, index) {
  const date = new Date(dateIn)
  date.setMonth(date.getMonth() + 1 + index)
  date.setDate(1) 
  return date
}

dates = new Array(basePayments).fill(new Date()).map(getDate)
```

Reply:
Makes sense, I've refactored as you suggest